core/src/main/scala/org/apache/spark/sql/delta/DeltaConfig.scala

```diff
@@ -508,10 +508,10 @@ trait DeltaConfigsBase extends DeltaLogging {
    * "stats_parsed" column. We will also write partition values as a struct as
    * "partitionValues_parsed".
    */
-  val CHECKPOINT_WRITE_STATS_AS_STRUCT = buildConfig[Option[Boolean]](
+  val CHECKPOINT_WRITE_STATS_AS_STRUCT = buildConfig[Boolean](
     "checkpoint.writeStatsAsStruct",
-    null,
-    v => Option(v).map(_.toBoolean),
+    "true",
```

Comment:
what are the implications of making it no longer optional?
(and if missing -- did it previously default to enabled or disabled?)

Reply:
Previously we used to check `checkpoint.writeStatsAsStruct` table property and if it is not set, then we used to look into DeltaSQLConf - `checkpointV2.enabled` which defaults to true.
So the net effect was "true" by default.

After this PR, there is no conf available and only `checkpoint.writeStatsAsStruct` table property is used to make a decision. This table property if not present will be considered as true. So the default behavior stays same ('true' by default).